lib/logstash/inputs/jdbc.rb

```diff
@@ -230,11 +233,37 @@ module LogStash module Inputs class Jdbc < LogStash::Inputs::Base
   config :target, :validate => :field_reference, :required => false
 
   attr_reader :database # for test mocking/stubbing
+  attr_reader :last_run_metadata_file_path # path to the file used as last run storage
 
   public
 
   def register
     @logger = self.logger
+
+    # This check is Logstash 5 specific.  If the class does not exist, and it
+    # won't in older versions of Logstash, then we need to set it to nil.
+    settings = defined?(LogStash::SETTINGS) ? LogStash::SETTINGS : nil
```

Comment:
similar issue here; LS5 is well past its end-of-life.

Reply:
Fixed by c3fc4008165d65438928fc32f72208d68eaf6ae3